### ISSUE TITLE
Implement AMP for most read container

### ISSUE THREAD

**Issue by thekp** [bbc/simorgh#6353]
**Is your feature request related to a problem? Please describe.**
Blocked on: https://github.com/ampproject/amphtml/issues/27251

In order to render most read with fresh data, we have to use AMP HTML components, in the case of most read <amp-list> (https://amp.dev/documentation/components/amp-list/?format=websites) is sufficient.

We created a proof of concept for using amp-list to render most read on our AMP pages.
PR: https://github.com/bbc/simorgh/pull/5935 from this PR, we found a bug with amp-list which has been noted down on this issue: https://github.com/ampproject/amphtml/issues/27251
The issue is that amp-list does not responsively resize itself so this causes extra spacing to stay intact when a user changes their viewport from small to large. 

AMP components functionality are run on the client-side, this means that we are unable to write Javascript that allows us to manipulate the fetched data from amp-list. We are restricted to their mustache templating (basic for loops and if statements). The only piece of data we require to manipulate is the `rank` value from the endpoint: https://www.bbc.com/pidgin/mostread.json
The most read component in psammead would manipulate the `listIndex` value and return a native numeral. However, this is not a MUST requirement for AMP most read.

**Describe the solution you'd like**
- POC here: https://github.com/bbc/simorgh/pull/6440
- We would also have to update most read in the psammead component to be able to accept the `rank` value from the endpoint as a string.
- We should ensure most-read styling is still consistent with our AMP variation.
- When we are happy with the above, we will also have to create a new route for `bbc.com/$service/popular/read.amp`

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Note: most read is currently being used on Article, Front, popular ((/popular/read)), and STY page.

We conducted some initial investigations into different approaches for most read on AMP, which can be seen on this issue:https://github.com/bbc/simorgh/issues/4742

TL:DR:
- We tried using amp-script but this was not possible to use with ReactJS as the library was too large.
- We decided with the product team to cut out some requirements of most read for AMP. e.g to not render the timestamp (show timestamp of promo if it's greater than 60days) and the freshness requirement (Do not render most read when the last updated value is greater than 35min).

**Comment by thekp** [comment]
**Initial investigation for implementation with `<amp-list>`:**
My PoC: https://github.com/bbc/simorgh/pull/6440

The PR linked above uses the `amp-list` AMP HTML component to create a container that renders most read on our WS front, article, and STY AMP pages (`.amp`).

In the PoC, `amp-list` is using an attribute layout of `responsive`, this enables the component to resize itself on a users' interaction (which includes clicking a button, scrolling on a page). In our case, most read is located at the bottom of the page, so a user would have to scroll down to see it and this will count as user interaction. However, if we render most read on its own page (/service/popular/read) then the component will not be at the bottom of the page and will not receive any user interaction so that it can resize itself.

**Problems with PoC:**
- Currently `amp-list` does not retract it's resizing so when a user adjusts their viewport from mobile to desktop. The most read list has a larger height on mobile and it will not dynamically reduce the height of the most read list when switched to a viewport larger than desktop. An example can be seen here: 
![ezgif com-video-to-gif (2)](https://user-images.githubusercontent.com/30599794/81206143-94703e00-8fc3-11ea-95ce-58615a9f8a6e.gif)

- We cannot do any data manipulation on the data that is fetched from the most read endpoint, this means that all services will get the same numeral ranking (1 to 10) and we cannot handle any timestamp requirements.

- Currently, most read component is configured to use native numerals for some services, this also means that they have specific spacing configured for them. Therefore if the most read component was to render with the numerals of 1 to 10 for these services, there may be spacing issues.

- `amp-list` renders items in a div element, so it will be difficult to apply `grid` related styling to the component as the grid will then see only one child element rather than multiple children elements. Hence the AMP most read container can only render most read in a single list column layout.

**Potential solutions:**
- We have already agreed with the product team that we are going to user server-side rendered data for the AMP most read page (`$service/popular/read.amp`). Therefore we shouldn't have to worry about using amp-list on the most read page.

- Render most-read on the associated page types and accept the fact following caveats:
1. the white space does not retract itself.
2. all services will get the 1 to 10 numerals.
3. most read is rendered in a single column layout for all viewports.


**Glossary**
_Responsive layout means: the element takes the space available to it and resizes its height automatically to the aspect ratio specified by the element's width and height attributes._

**Comment by DarioR01** [comment]
Following the previous investigation, we managed to implement a working client-side version of the most-read AMP [here](https://github.com/bbc/simorgh/pull/10017). 
We did not pursue server-side render as suggested by the original investigation as the amp cache could theoretically not meet the 35 min update requirement.
This [documentation](https://developers.google.com/amp/cache/overview#google-amp-cache-updates) about Amp cache states:
> When a user makes a request for something that is stale, that request causes a new copy to be fetched, so that the next user gets fresh content. 

This implies that some users will need to frequently visit a page in order for it to update using server-side rendering, with the risk that the page could become stale.

**Comment by DarioR01** [comment]
Investigation AMP for client-side: https://github.com/bbc/simorgh/pull/10017
